test/e2e/harness.go

```diff
@@ -17,33 +18,58 @@ import (
 	"io/ioutil"
 	"math/rand"
 	"net/http"
+	"net/url"
 	"os"
 	"path/filepath"
+	"strconv"
 	"time"
 )
 
 type E2EConfig struct {
-	Bootstrap   bool
-	ApiEndpoint string
+	bootstrap   bool
+	apiEndpoint string
 	baseUrl     string
+
+	stressTest StressTestConfig
+}
+
+type StressTestConfig struct {
+	numberOfTransactions  int64
+	acceptableFailureRate int64
+	targetTPS             float64
 }
 
 const LOCAL_NETWORK_SIZE = 3
 const START_HTTP_PORT = 8090
 
 func getConfig() E2EConfig {
-	Bootstrap := len(os.Getenv("API_ENDPOINT")) == 0
-	baseUrl := fmt.Sprintf("http://localhost:%d", START_HTTP_PORT + 2) // 8080 is leader, 8082 is node-3
-	ApiEndpoint := fmt.Sprintf("%s/api/v1/", baseUrl)
-
-	if !Bootstrap {
-		ApiEndpoint = os.Getenv("API_ENDPOINT")
+	bootstrap := len(os.Getenv("API_ENDPOINT")) == 0
+	baseUrl := fmt.Sprintf("http://localhost:%d", START_HTTP_PORT+2) // 8080 is leader, 8082 is node-3
+	apiEndpoint := fmt.Sprintf("%s/api/v1/", baseUrl)
+
+	stressTestNumberOfTransactions := int64(10000)
+	stressTestFailureRate := int64(2)
+	stressTestTargetTPS := float64(700)
+
+	if !bootstrap {
+		apiEndpoint = os.Getenv("API_ENDPOINT")
+		url, _ := url.Parse(apiEndpoint)
+		baseUrl = url.Scheme + "://" + url.Host
+
+		stressTestNumberOfTransactions, _ = strconv.ParseInt(os.Getenv("STRESS_TEST_NUMBER_OF_TRANSACTIONS"), 10, 0)
```

Comment:
I don't really see why we need to propagate these values from the outside

Reply:
Because they are different for different conditions. We might want to check how system behaves under 100, 1000, 10000, 100000 tx and so on.